src/js/components/InfiniteScroll/InfiniteScroll.js

```diff
@@ -147,6 +147,14 @@ const InfiniteScroll = ({
     }
   }, [items.length, lastPage, onMore, pendingLength, renderPageBounds, step]);
 
+  useEffect(() => {
+    if (lastPage === 0 && pendingLength !== 0) {
+      setPageHeights([]);
+      setPendingLength(0);
+      setRenderPageBounds([0, show ? Math.floor((show + step) / step) - 1 : 0]);
```

Comment:
I think this is fine functionally. One small improvement would be to create something like:
`const calculateLastPage = (show, step) => (show ? Math.floor((show + step) / step) - 1 : 0);`
outside of the component in this file and then call it in the three places we have that expression.

Reply:
Done!
Went with `calculateLastPageBound` as we already have a `lastPage` variable that isn't directly related to this function.